### PR TITLE
python311Packages.tinytuya: init at 0.1.51

### PR DESCRIPTION
**Pull request by pathob**
## Description of changes

[tinytuya](https://pypi.org/project/tinytuya/) is a Python module for working with Tuya-powered IoT devices. It is actively maintained, provides an own CLI and is a dependency of the Home Assistant custom component "[tuya-local](https://github.com/make-all/tuya-local)" that I want to add in a next step (not to confuse with with "[localtuya](https://github.com/rospogrigio/localtuya)" that is already part of nixpkgs).

This is my first contribution to NixOS nixpkgs and I feel like I've done everything that is required for this small change, even though not many boxes are ticket. If something needs to be done on my end, please let me know. Thanks!

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/tinytuya/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+, fetchFromGitHub
+, cryptography
+, requests
+, colorama
+}:
+
+buildPythonPackage rec {
+  pname = "tinytuya";
+  version = "1.13.2";
+  format = "setuptools";
```
Comment (fabaff) [inline pkgs/development/python-modules/tinytuya/default.nix:12]:
```suggestion
  pyproject = true;
```

and then `build-system = [ setuptools ];`. See https://nixos.org/manual/nixpkgs/stable/#python

File: pkgs/development/python-modules/tinytuya/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+  src = fetchFromGitHub {
+    owner = "jasonacox";
+    repo = "${pname}";
```
Comment (fabaff) [inline pkgs/development/python-modules/tinytuya/default.nix:16]:
```suggestion
    repo = "tinytuya";
```

I think that there is still no general consents about this but inside the Python eco-system usually we go the route with `repo = "repo-name";`

File: pkgs/development/python-modules/tinytuya/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    rev = "refs/tags/v${version}";
+    hash = "sha256-44x5P+Ej/d6B5n53iDuLDBzkeZZvArpcgzXLJBcIJe0=";
+  };
+
+  propagatedBuildInputs = [
```
Comment (fabaff) [inline pkgs/development/python-modules/tinytuya/default.nix:21]:
```suggestion
  dependencies = [
```

File: pkgs/development/python-modules/tinytuya/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    cryptography
+    requests
+    colorama
+  ];
+
+  # Tests require real network resources
+  doCheck = false;
+
+  meta = with lib; {
+    description = "Python API for Tuya WiFi smart devices using a direct local area network (LAN) connection or the cloud (TuyaCloud API).";
```
Comment (fabaff) [inline pkgs/development/python-modules/tinytuya/default.nix:31]:
```suggestion
    description = "Python API for Tuya WiFi smart devices using a direct local area network (LAN) connection or the cloud (TuyaCloud API)";
```

No period at the end please.

File: pkgs/development/python-modules/tinytuya/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    homepage = "https://github.com/jasonacox/tinytuya";
+    changelog = "https://github.com/jasonacox/tinytuya/releases/tag/v${version}";
+    license = licenses.mit;
+    maintainers = [ ];
```
Comment (fabaff) [inline pkgs/development/python-modules/tinytuya/default.nix:35]: New packages need a maintainer. Add yourself to `[maintainers/maintainer-list.nix](https://github.com/NixOS/nixpkgs/blob/master/maintainers/maintainer-list.nix)` in a separate commit then you can add your handle here.

**Comment by pathob** [comment]
Hi @fabaff, thanks for your review, I've implemented all suggestions.

**Review by natsukium — CHANGES_REQUESTED** [review]

**Comment by pathob** [comment]
Hi natsukium, thanks for your review too, all done.

**Review by natsukium — APPROVED** [review]
LGTM, thanks for your contribution!
Welcome to nixpkgs!🎉

**Pull request merged**